02_bubble/runnerdave/main_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote(`[1 2 3 5]`)
+	actual := strconv.Quote(buf.String())
+	t.Logf("expected:%s", expected)
+	t.Logf("actual:%s", actual)
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()")
+	}
+}
+
+func TestSort(t *testing.T) {
+	tables := []struct {
+		x []int
+		n []int
+	}{
+		{[]int{1, 3, 2, 5}, []int{1, 2, 3, 5}},
```

Comment:
Can you add tests for edge cases: empty and 1 element slices.
You should also test that the sort does not modify the original slice

Reply:
done